### PR TITLE
theme as peer dep

### LINKED ISSUE #604: Theme dep management
In dev only, two theme component versions, problem with light theme.

Maybe move peerDep.

Resolved by bundler in stage/prod.

Theory: react 16 context creates multiple instances if multiple theme packages exist.  Inner component sees the other context that is not rendered and defaults (falsely) to dark.

### PR DESCRIPTION
**Pull request by danethurber**
### What You're Solving

- resolves #604 

### Design Decisions

- packages that need theming now have the theme package as a peerDependency
- added theme devDep to packages for easier to find version requirements from lerna versioning
- updated CRA example to test package version publishing and resolution

### CODE REVIEW

File: packages/site/pages/install.js [hunk 1]
```diff
@@ -25,7 +25,13 @@ Use of the Core or Components requires the existence of the Design System normal
 npm install @pluralsight/ps-design-system-normalize
 \`\`\`
 
-Include this vanilla CSS in your application in a method appropriate for your project. For usage options, follow the [Core Usage docs](/core/usage) patterns.
```
Comment (jaketrent) [inline packages/site/pages/install.js]: Why remove this?  Still seems valid (ie, all components still require normalize)

File: scripts/create-component/package-template/package.json [hunk 2]
```diff
@@ -22,6 +22,7 @@
   "dependencies": {{dependencies}},
   "peerDependencies": {
     "@pluralsight/ps-design-system-normalize": "*",
+    "@pluralsight/ps-design-system-theme": "*",
```
Comment (jaketrent) [inline scripts/create-component/package-template/package.json]: Do these peerDeps w/ versions of "*" get published that way, or does it publish with a concrete (eg, "3.0.4") version?

File: packages/site/pages/install.js [hunk 3]
```diff
@@ -25,7 +25,13 @@ Use of the Core or Components requires the existence of the Design System normal
...
+## Theming
+
+Most components support multiple themes and have a dependency on theme package. Install with:
```
Comment (jaketrent) [inline packages/site/pages/install.js]: I would modify this.  We've tried hard to make sure that *all* components support theming.  Whether there's an actual diff between themes is less important.  It's the perception problem that I don't want to reawaken.

Maybe:

```
All components support theming and have a peer dependency on the Theme package. Install with:
```

I would also *move* this down to the "Use Components" section.

File: packages/site/pages/install.js (hunk 1 above)
Comment (danethurber) [reply packages/site/pages/install.js]: oops

File: scripts/create-component/package-template/package.json (hunk 2 above)
Comment (danethurber) [reply scripts/create-component/package-template/package.json]: doesn't look like lerna does the versioning for us 😢 

i'll go through and update

File: packages/viewtoggle/package.json [hunk 4]
```diff
@@ -26,7 +26,7 @@
     "prop-types": "^15.7.2"
   },
   "peerDependencies": {
-    "@pluralsight/ps-design-system-normalize": "*",
+    "@pluralsight/ps-design-system-normalize": "v3.1.4",
```
Comment (jaketrent) [inline packages/viewtoggle/package.json]: I'm pretty sure these numbers don't need and can't have the `v` prefix.  Do you know otherwise?

Comment (danethurber) [reply packages/viewtoggle/package.json]: oops. it should be a ^

File: scripts/create-component/package-template/package.json [hunk 5]
```diff
@@ -21,7 +21,8 @@
   },
   "dependencies": {{dependencies}},
   "peerDependencies": {
-    "@pluralsight/ps-design-system-normalize": "*",
+    "@pluralsight/ps-design-system-normalize": "v3.1.4",
+    "@pluralsight/ps-design-system-theme": "v3.0.2",
```
Comment (jaketrent) [inline scripts/create-component/package-template/package.json]: rm v

File: packages/theme/package.json [hunk 6]
```diff
@@ -20,7 +20,7 @@
     "prop-types": "^15.7.2"
   },
   "peerDependencies": {
-    "@pluralsight/ps-design-system-normalize": "*",
+    "@pluralsight/ps-design-system-normalize": "v3.1.4",
```
Comment (jaketrent) [inline packages/theme/package.json]: rm v

**Pull request merged**